blaze/server/server.py

```diff
@@ -607,14 +607,22 @@ def addserver(payload, serial):
         return (error_msg % list(payload.keys()),
                 RC.UNPROCESSABLE_ENTITY)
 
-    [(name, resource_uri)] = payload.items()
+    [(name, resource_info)] = payload.items()
 
     if name in data:
         msg = "Cannot add dataset named %s, already exists on server."
         return (msg % name, RC.CONFLICT)
 
     try:
-        data.update({name: resource(resource_uri)})
+        if isinstance(resource_info, dict):
+            # Extract resource creation arguments
+            source = resource_info['source']
+            extra = resource_info.get('extra', {})
```

Comment:
It's an edge case, but some `resource()` implementations in odo do take both positional and kwargs, and treat the positional args in a special way.  For instance:
    https://github.com/blaze/odo/blob/master/odo/backends/sql.py#L545

I also realize that the original code didn't handle this case, but this is an opportunity to do it correctly :)

Can you refactor this so that the `resource_info` has both an `args` sequence and a `kwargs` dict / mapping, ala:

```python
source = ...
args = resource_info.get('args', [])
kwargs = resource_info.get('kwargs', {})
...
data.update({name: resource(source, *args, **kwargs)})
```